fp3/tools.py

```diff
@@ -0,0 +1,101 @@
+from dials.array_family import flex
+from dxtbx.model.experiment_list import ExperimentList
+
+
+def combine_reflections(fins, fout):
+    """Combine reflection files from fins into a single output file fout.
+    Makes assumptions that the data are consistent from one to the next, no
+    particular tests are performed at the moment."""
+
+    d0 = flex.reflection_table.from_file(fins[0])
+
+    for f in fins[1:]:
+        d1 = flex.reflection_table.from_file(f)
+
```

Comment:
Testing on the x4-wide dataset, I also got a failure due to dividing by zero variance at L61. If I remember correctly, you can get a zero variance set if the reflections were not integrated.
I think all that's needed to clean the data is this:


```suggestion
    d0 = flex.reflection_table.from_file(fins[0])
    if "intensity.prf.variance" in d0:
        d0 = d0.select(d0["intensity.prf.variance"] > 0)
    d0 = d0.select(d0["intensity.sum.variance"] > 0)

    for f in fins[1:]:
        d1 = flex.reflection_table.from_file(f)
        if "intensity.prf.variance" in d1:
            d1 = d1.select(d1["intensity.prf.variance"] > 0)
        d1 = d1.select(d1["intensity.sum.variance"] > 0)

```